src/nodes/chunk_dispatch/chunk_insert_state.c

```diff
@@ -122,6 +122,9 @@ create_chunk_rri_constraint_expr(ResultRelInfo *rri, Relation rel)
 	}
 }
 
+#undef makeNode
+#define makeNode(_type_) ((_type_ *) newNode(2 * sizeof(_type_), T_##_type_))
```

Comment:
Yupp, this is indeed horrible, as you put it.

Reply:
Why do we need it? The palloc code alloced 512 byte chunks, which should be enough as explained here: https://www.postgresql.org/message-id/1972733.1731612633%40sss.pgh.pa.us